src/TCommandLine.cpp

```diff
@@ -862,17 +842,24 @@ void TCommandLine::enterCommand(QKeyEvent* event)
     }
 
     if (!toPlainText().isEmpty()) {
-        mHistoryBuffer = 0;
+        if (mpHost->mAutoClearCommandLineAfterSend) {
+            mHistoryBuffer = 0;
+        } else {
+            mHistoryBuffer = 1;
+        }
         setPalette(mRegularPalette);
 
         mHistoryList.removeAll(toPlainText());
-        mHistoryList.push_front(toPlainText());
+        if (!mHistoryList.isEmpty()) {
+            mHistoryList[0] = toPlainText();
+        } else {
+            mHistoryList.push_front(toPlainText());
+        }
+        mHistoryList.push_front("");
```

Comment:
It is cleaner to use `QString()`  rather than an empty raw string literal to specify an empty `QString`...

Reply:
Fixed.